### ISSUE TITLE
path.reverseNodes bug

### ISSUE THREAD

**Issue by glemmon** [neo4j/neo4j#5086]
I am using Neo4j 2.2.1. I am trying to use the reverseNodes method of the path class. Running this code:

``` java
logger.info("Path1: "+path.toString());
for(Node node : path.reverseNodes()){
   logger.info("Node1: "+node.toString());
}
```

Logger output:

``` java
INFO: Path1: (469383)--[is_a,722474]-->(469384)--[is_a,732353]-->(469382)--[is_a,722473]-->(469089)--[is_a,722737]-->(468447)--[is_a,722237]-->(467769)--[is_a,721025]-->(467627)
```

We never get inside the for loop. An NotFoundException is thrown first:

``` java
message: "Node[467627J not connected to this relationship[722474]"
exception: "NotFound.Exception"
fullname: "org.neo4j.graphdb.NotFound.Exception"
stackTrace:
0: "org.neo4j.kernel.impl.core.RelationshipProxy.getOtherNode(RelationshipProxy.java:189)'
1: "org. neo4j. graphalgo. impl. util. Pathlmpl$1$1. next(Pathlmpl. java: 205)"
2: "org. neo4j. graphalgo. impl. util. Pathlmpl$1$1. next(Pathlmpl. java: 187)"
3: "org.eihg.phevor.plugins.TextQuery.path_to_json(TextQuery.java:85)"
```

It looks as though we are starting from the end node but trying to go through the first relationship instead of the last.

Seems like a bug in Neo4j. Am I missing something here?

**Comment by jakewins** [comment]
Do you have a full stack trace?

**Comment by chrisvest** [comment]
This was fixed in 2.2.5 by #5130
